assets/js/components/PresenterPopup.tsx

```diff
@@ -7,17 +7,30 @@ import {
   useDisclosure,
 } from "@chakra-ui/react";
 import React from "react";
+import type { PresenterPopupState } from "../pages/Event";
 
-type PopupProp = {
-  onAccept: Function;
-  onReject: Function;
+type PresenterPopupProps = {
+  name: string;
+  moderator: string;
+  eventChannel: any;
+  setPopupState: React.Dispatch<React.SetStateAction<PresenterPopupState>>;
 };
 
-const PresenterPopup = ({ onAccept, onReject }: PopupProp) => {
+const PresenterPopup = ({ name, moderator, eventChannel, setPopupState }: PresenterPopupProps) => {
   const { isOpen, onClose } = useDisclosure({ defaultIsOpen: true });
 
+  const onReject = () => {
+    eventChannel.push("presenter_answer", { name: name, moderator: moderator, answer: "reject" });
+    setPopupState({ isOpen: false, moderator: "" });
+  };
+
+  const onAccept = () => {
+    eventChannel.push("presenter_answer", { name: name, moderator: moderator, answer: "accept" });
+    setPopupState({ isOpen: false, moderator: "" });
+  };
```

Comment:
Could you move this to a separate function, which will take only the parameter `answer`?